### PR TITLE
tools, add azure-rest-api-specs-examples-automation

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
migrate from https://github.com/weidongxu-microsoft/azure-rest-api-specs-examples-automation

### CODE REVIEW

File: tools/azure-rest-api-specs-examples-automation/ci.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+schedules:
+- cron: "0 12 * * *"
+  displayName: Daily build
+  branches:
+    include:
+    - main
+  always: true
+
+variables:
+- group: Release Secrets for GitHub
```
Comment (weshaggard) [inline tools/azure-rest-api-specs-examples-automation/ci.yml:14]: We generally hook these up on the DevOps side to abstract the variable groups.

File: tools/azure-rest-api-specs-examples-automation/ci.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+trigger: none
+
+pr: none
+
+schedules:
```
Comment (weshaggard) [inline tools/azure-rest-api-specs-examples-automation/ci.yml:5]: We have typically done the schedules on the DevOps side but I don't mind if you wnat to do this here as well.

File: tools/azure-rest-api-specs-examples-automation/ci.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+trigger: none
```
Comment (weshaggard) [inline tools/azure-rest-api-specs-examples-automation/ci.yml:1]: Should we have this trigger on changes in this directory or do you want another pipeline that will verify the tool itself?

File: tools/azure-rest-api-specs-examples-automation/ci.yml (hunk 2 above)
Comment (weidongxu-microsoft) [reply tools/azure-rest-api-specs-examples-automation/ci.yml]: Sure. I removed `schedules` and `variables` from the ci.yml. Would configure them in DevOps.

File: tools/azure-rest-api-specs-examples-automation/ci.yml (hunk 3 above)
Comment (weidongxu-microsoft) [reply tools/azure-rest-api-specs-examples-automation/ci.yml:1]: Yes, I should add a CI for this package as well (run the tests etc.).

Should we name that verification CI `ci.yml`, and rename this "automation of collect samples" pipeline to e.g. `ci-automation.yml`?

File: tools/azure-rest-api-specs-examples-automation/ci.yml (hunk 2 above)
Comment (weshaggard) [reply tools/azure-rest-api-specs-examples-automation/ci.yml]: Probably outside of usual redmond and shanghai core business hours.

File: tools/azure-rest-api-specs-examples-automation/ci.yml (hunk 3 above)
Comment (weshaggard) [reply tools/azure-rest-api-specs-examples-automation/ci.yml:1]: Yes I'd name the verification ci.yml and the automation just "samples-automation.yml" and that name should also be included as part of the pipeline name in devops under the automation folder.

Comment (weidongxu-microsoft) [reply tools/azure-rest-api-specs-examples-automation/ci.yml]: Got it.

Add CI as `tools - azure-rest-api-specs-examples-automation - ci`, public, tools folder
Add automation as `automation - samples-automation`, internal, automation folder

**Pull request merged**